### PR TITLE
Adding xcconfig files for missing projects

### PR DESCRIPTION
**Pull request by bhariharan**
Added xcconfig files for:

* ```SalesforceHybridSDK```
* ```SalesforceRestAPI```
* ```SmartSync```

### CODE REVIEW

**Comment by wmathurin** [comment]
:+1:

File: libs/SalesforceHybridSDK/SalesforceHybridSDK-Common.xcconfig [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+//  Created by Bharath Hariharan on 4/7/15.
+//  Copyright (c) 2015 salesforce.com. All rights reserved.
+//
+
+GCC_PRECOMPILE_PREFIX_HEADER = YES
+GCC_PREFIX_HEADER = $(PROJECT_NAME)/$(PROJECT_NAME)-Prefix.pch
+
+PRODUCT_NAME = SalesforceHybridSDK
+
+HEADER_SEARCH_PATHS = "$(BUILT_PRODUCTS_DIR)/Headers" "$(TARGET_BUILD_DIR)/Headers" "$(PROJECT_DIR)/../../external/cordova-ios/cordova-ios/CordovaLib/Classes" "$(PROJECT_DIR)/../../external/ThirdPartyDependencies/SalesforceCommonUtils/Headers" $(inherited)
```
Comment (khawkins) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK-Common.xcconfig]: The CordovaLib/Classes may need a `/**` at the end, if it's supposed to be recursive (which I think it is).

**Comment by khawkins** [comment]
Looks fine otherwise.  Do we want to remove the old config values, like `HEADER_SEARCH_PATHS`, from the project files?

File: libs/SalesforceHybridSDK/SalesforceHybridSDK-Common.xcconfig (hunk 1 above)
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK-Common.xcconfig]: You're right. I'll add that in.

**Comment by bhariharan** [comment]
Should be good to go now.

**Comment by khawkins** [comment]
Cool, looks good.

**Pull request merged**